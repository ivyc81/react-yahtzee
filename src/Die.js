import React, { Component } from "react";
import "./Die.css";


/**
 * presentational component
 * display value
 * diaplay according to locked or not
 */

class Die extends Component {
  render() {
    return (
      <button
        className={this.props.locked ? "Die Die-locked" : "Die"}
        onClick={() => this.props.handleClick(this.props.idx)}>
        {this.props.val}
      </button>
    );
  }
}

export default Die;
