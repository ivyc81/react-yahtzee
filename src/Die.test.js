import React from 'react';
import { shallow, mount } from "enzyme";
import Die from './Die';
import toJson from "enzyme-to-json";

it('renders without crashing', () => {
  shallow(<Die />);
});

it('renders correctly unlocked', () => {
    const wrapper = mount(<Die locked={ false }/>);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it('renders correctly locked', () => {
    const wrapper = mount(<Die locked={ true }/>);
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

