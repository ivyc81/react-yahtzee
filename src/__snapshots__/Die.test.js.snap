// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly locked 1`] = `
<Die
  locked={true}
  val={2}
>
  <button
    className="Die Die-locked"
    onClick={[Function]}
  >
    2
  </button>
</Die>
`;

exports[`renders correctly unlocked 1`] = `
<Die
  locked={false}
  val={2}
>
  <button
    className="Die"
    onClick={[Function]}
  >
    2
  </button>
</Die>
`;
